test/unit/knex.js

```diff
@@ -285,11 +284,10 @@ describe('knex', () => {
 
     const knexWithParams = knex.withUserParams({ userParam: '451' });
 
-    return knexWithParams.transaction((trx) => {
+    return knexWithParams.transaction(async (trx) => {
       expect(trx.userParams).to.deep.equal({
         userParam: '451',
       });
-      return bluebird.resolve();
```

Comment:
I see.  So, this was to work around the `Transaction` class's special handling for callbacks that return Promises.  (Otherwise, the `Transaction` class would assume that the caller is responsible for committing / rolling back the transaction)